app/lightning.js

```diff
@@ -4,53 +4,109 @@ const fs = require("fs");
-							+ "Please add the missing lnd macaroon file or update/remove the path in the application configuration.");
-					process.exit(1);
-				}
-			} else {
-				credentials = sslCreds;
-			}
+        return credentials;
+    }
 
-			return new lnrpcDescriptor.lnrpc.Lightning(lndHost, credentials);
+    /**
+     * @constructor
+     * @param {string} protoPath - the path to the `rpc.proto` file that defined the `Lightning` RPC interface
+     * @param {string} lndHost - the host and port of the LND node (ex. "locahost:10003")
+     * @param {string} lndCertPath - the path to the SSL certificate used by LND 
+     * @param {?string} macaroonPath - the path to the macarron file to use. Can be `null` if no macaroon should be used.
+     */
+    constructor(protoPath, lndHost, lndCertPath, macaroonPath) {
+        process.env.GRPC_SSL_CIPHER_SUITES = "HIGH+ECDSA";
 
-		} else {
+	if (!fs.existsSync(lndCertPath)) {
+            logger.error("Required lnd certificate path missing from application configuration.");
+            process.exit(1);
+        }
 
-			logger.error("The specified lnd certificate file \"" + lndCertPath + "\" was not found.\n"
-					+ "Please add the missing lnd certificate file or update the path in the application configuration.");
-			process.exit(1);
+        // Define credentials for SSL certificate generated by LND, and active client
+        this.lndHost = lndHost;
+	this.lndCert = fs.readFileSync(lndCertPath);
+        this.lnrpcDescriptor = grpc.load(protoPath);
+	this.macaroonPath = macaroonPath;
+        this.activeClient = null;
+    }
 
-		}
+    /*
+     * Calls a Lightning gRPC method.
+     * @param {string} method - the gRPC method to call (ex. "getInfo")
+     * @param {Object} parameters - optional key/value parameters to supply for the API call
+     * @returns {Promise} - if successful, response if an Object containing API result payload, otherwise it will fail
+       with a LightningError.
+     */
+    async call(method, parameters) {
+        return new Promise((resolve, reject) => {
+            let activeClient = this.getActiveClient();
+            activeClient[method](parameters, (err, response) => {
+                if (err) {
 
-	} else {
+                    // drop active client, so that it can be recreated
+                    this.activeClient = null;
 
-		logger.error("Required lnd certificate path missing from application configuration.");
-		process.exit(1);
+                    switch(err.code) {
+                        case grpc.status.UNIMPLEMENTED:
+                            reject(LightningError.WALLET_LOCKED);
+                            break;
+                        case grpc.status.UNAVAILABLE:
+                            reject(LightningError.NODE_UNREACHABLE);
+                            break;
+                        default:
+                            logger.error("Unrecognized gRPC error: " + err);
+                            reject(LightningError.UNCATEGORIZED);
```

Comment:
@PirosB3 looks like we are losing the original gRPC error here.
We need it to be able to handle it correctly on the web-client side.